modules/noise/noise_texture.cpp

```diff
@@ -143,18 +155,42 @@ Ref<Image> NoiseTexture::_generate_texture() {
 	Ref<Image> image;
 
 	if (seamless) {
-		image = ref_noise->get_seamless_image(size.x, size.y, invert, seamless_blend_skirt);
+		image = ref_noise->get_seamless_image(size.x, size.y, invert, in_3d_space, seamless_blend_skirt);
 	} else {
-		image = ref_noise->get_image(size.x, size.y, invert);
+		image = ref_noise->get_image(size.x, size.y, invert, in_3d_space);
+	}
+	if (color_ramp.is_valid()) {
+		image = _modulate_with_gradient(image, color_ramp);
 	}
-
 	if (as_normal_map) {
 		image->bump_map_to_normal_map(bump_strength);
 	}
+	if (generate_mipmaps) {
+		image->generate_mipmaps();
+	}
 
 	return image;
 }
 
+Ref<Image> NoiseTexture::_modulate_with_gradient(Ref<Image> p_image, Ref<Gradient> p_gradient) {
+	int width = p_image->get_width();
+	int height = p_image->get_height();
+
+	Ref<Image> new_image;
+	new_image.instantiate();
+	new_image->create(width, height, false, Image::FORMAT_RGBA8);
+
+	for (int row = 0; row < height; row++) {
+		for (int col = 0; col < width; col++) {
+			Color pixel_color = p_image->get_pixel(col, row);
+			Color ramp_color = color_ramp->get_color_at_offset(pixel_color.get_luminance());
+			new_image->set_pixel(col, row, ramp_color);
+		}
+	}
```

Comment:
```suggestion
	Vector<uint8_t> data;
	data.resize(width * height * 4);

	uint8_t *wd8 = data.ptrw();

	for (int row = 0, x = 0; row < height; row++) {
		for (int col = 0; col < width; col++, x++) {
			Color pixel_color = p_image->get_pixel(col, row);
			Color ramp_color = color_ramp->get_color_at_offset(pixel_color.get_luminance());
			wd8[x * 4 + 0] = uint8_t(CLAMP(ramp_color.r * 255, 0, 255));
			wd8[x * 4 + 1] = uint8_t(CLAMP(ramp_color.g * 255, 0, 255));
			wd8[x * 4 + 2] = uint8_t(CLAMP(ramp_color.b * 255, 0, 255));
			wd8[x * 4 + 3] = uint8_t(CLAMP(ramp_color.a * 255, 0, 255));
		}
	}

	new_image->create(width, height, false, Image::FORMAT_RGBA8, data);
```

This is derived from the `FastNoiseLite::get_image()` function that was removed. It should be more performant than using `Image.set_pixel()` directly. I haven't tested this code, but it should work (or at least be close to working).

Reply:
You're right, the code you suggested would be slightly faster, but the problem is the lack of flexibility. (a Noise resource could generate a float texture). NoiseTexture should be able to handle all image formats because it has no information about which format `get_image` or `get_seamless_image` returns.